### PR TITLE
mrc-1963 - Add indication that a project was shared by someone else

### PR DESCRIPTION
**Pull request by carbowizzy**
## Description

This PR adds indication that a project was shared by someone else, a user is able to see the email address of a user whose project was shared, this email address should be able to link back to the user as soon as we have a profile page for users.

## Type of version change
_Delete as appropriate_

Minor 

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/store/translations/locales.ts [hunk 1]
```diff
@@ -500,6 +502,7 @@ const fr: Partial<Translations> = {
     sex: "Sexe",
     shape: "Fichier de formes",
     share: "Partager",
+    sharedBy: "partagé par",
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/translations/locales.ts:505]: I think we'll want to make this capitalised: "Partagé par"

File: src/app/static/src/tests/components/projects/projectHistory.test.ts [hunk 2]
```diff
@@ -154,6 +154,7 @@ describe("Project history component", () => {
         expectTranslated(headers.at(6), "Delete", "Supprimer", store);
         expectTranslated(headers.at(7), "Copy to", "Copier", store);
         expectTranslated(headers.at(8), "Share", "Partager", store);
+        expectTranslated(headers.at(9), "SharedBy", "partagé par", store);
```
Comment (EmmaLRussell) [inline src/app/static/src/tests/components/projects/projectHistory.test.ts:157]: Need a test for the shared by value too

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]
This is really neat and works great! :+1: 

I think it would be even nicer if the 'Shared by' column was not visible if the user does not have any projects shared with them, as will probably be the case for most users. Should just be a case of adding a computed property `anySharedProjects` to the component, which returns a count of the filtered project list where sharedBy is truthy, and `v-if`-ing the relevant elements on that. In that case we could donate the extra column width to the project name  column. Happy for that to go into a separate ticket though, if you want to add one to the sprint.

File: src/app/static/src/app/store/translations/locales.ts [hunk 3]
```diff
@@ -321,6 +322,7 @@ const en: Translations = {
     sex: "Sex",
     shape: "Area boundary file",
     share: "Share",
+    sharedBy: "SharedBy",
```
[the remark this answers is not in the document]
Comment (carbowizzy) [reply src/app/static/src/app/store/translations/locales.ts]: I actually tried to manage the space as the text gets truncated.

File: src/app/src/main/kotlin/org/imperial/mrc/hint/models/Project.kt [hunk 4]
```diff
@@ -3,7 +3,10 @@ package org.imperial.mrc.hint.models
 import java.sql.Timestamp
 import java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME
 
-data class Project(val id: Int, val name: String, val versions: List<Version>)
+data class Project (val id: Int, val name: String, val sharedBy: String, val versions: List<Version>)
+{
+    constructor(id: Int, name: String, versions: List<Version>) : this( id,  name, "", versions)
```
[the remark this answers is not in the document]
Comment (carbowizzy) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/models/Project.kt]: You are right, I tried to use the optional param but wasn't working so I abandoned it. I figured why it didn't work, it's now refactored.

File: src/app/src/main/kotlin/org/imperial/mrc/hint/db/ProjectRepository.kt [hunk 5]
```diff
@@ -71,12 +73,23 @@ class JooqProjectRepository(private val dsl: DSLContext) : ProjectRepository
         return result[PROJECT.ID]
     }
 
+    override fun saveClonedProject(userId: String, projectName: String, sharedBy: String): Int
```
[the remark this answers is not in the document]
Comment (carbowizzy) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/db/ProjectRepository.kt]: I will create a new ticket to get the column hidden when not in use.

**Review by EmmaLRussell — APPROVED** [review]

**Comment by EmmaLRussell** [comment]
Looks good - could you just fix the version conflict then go ahead and merge.

**Pull request merged**